Meta/gn/secondary/Meta/Lagom/Contrib/MacPDF/BUILD.gn

```diff
@@ -0,0 +1,104 @@
+import("//Meta/gn/build/mac/compile_xib_resources.gni")
+
+assert(current_os == "mac", "This application is macOS specific")
+
+group("MacPDF") {
+  deps = [ ":MacPDF.app" ]
+}
+
+compile_xib_resources("pdf_nibs") {
+  sources = [ "MainMenu.xib" ]
+}
+
+executable("MacPDF_executable") {
+  include_dirs = [
+    "//Userland/Libraries",
+    "//Userland/Services",
+  ]
+  defines = [ "AK_DONT_REPLACE_STD" ]
+  cflags_objcc = [ "-fobjc-arc" ]
+  deps = [
+    ":pdf_nibs",
+    "//AK",
+    "//Userland/Libraries/LibCore",
+    "//Userland/Libraries/LibGfx",
+    "//Userland/Libraries/LibPDF",
+  ]
+  frameworks = [
+    "Cocoa.framework",
+    "UniformTypeIdentifiers.framework",
+  ]
+  sources = [
+    "AppDelegate.mm",
+    "MacPDFDocument.mm",
+    "MacPDFOutlineViewDataSource.mm",
+    "MacPDFView.mm",
+    "MacPDFWindowController.mm",
+    "main.mm",
+  ]
+  output_name = "MacPDF"
+}
+
+bundle_data("pdf_bundle_info_plist") {
+  sources = [ "Info.plist" ]
+  outputs = [ "{{bundle_contents_dir}}/Info.plist" ]
+}
+
+bundle_data("pdf_bundle_executables") {
+  public_deps = [ ":MacPDF_executable" ]
+  sources = [ "$root_out_dir/bin/MacPDF" ]
+  outputs = [ "{{bundle_executable_dir}}/{{source_file_part}}" ]
+}
+
+bundle_data("pdf_bundle_libs") {
+  public_deps = [
+    "//Userland/Libraries/LibCompress",
+    "//Userland/Libraries/LibCore",
+    "//Userland/Libraries/LibCrypto",
+    "//Userland/Libraries/LibFileSystem",
+    "//Userland/Libraries/LibGfx",
+    "//Userland/Libraries/LibIPC",
+    "//Userland/Libraries/LibPDF",
+    "//Userland/Libraries/LibTextCodec",
+  ]
+  sources = [
+    "$root_out_dir/lib/liblagom-compress.dylib",
+    "$root_out_dir/lib/liblagom-core.dylib",
+    "$root_out_dir/lib/liblagom-crypto.dylib",
+    "$root_out_dir/lib/liblagom-filesystem.dylib",
+    "$root_out_dir/lib/liblagom-gfx.dylib",
+    "$root_out_dir/lib/liblagom-ipc.dylib",
+    "$root_out_dir/lib/liblagom-pdf.dylib",
+    "$root_out_dir/lib/liblagom-textcodec.dylib",
+  ]
+  outputs = [ "{{bundle_contents_dir}}/lib/{{source_file_part}}" ]
+}
+
+bundle_data("pdf_bundle_nibs") {
+  public_deps = [ ":pdf_nibs" ]
+  sources = get_target_outputs(":pdf_nibs")
+  outputs = [ "{{bundle_resources_dir}}" + "{{source_file_part}}" ]
+}
+
+bundle_data("pdf_resources") {
+  # FIXME: Don't list directories here, list out the files directly
```

Comment:
At the moment it's just 4 files here, might as well do that instead of adding a fix me?

Reply:
Four files? I'm not sure that's accurate. When you instantiate the Gfx::FontDatabase, we load every resource in the font/ path. Does LibPDF load 4 specific fonts? I couldn't find any use of it in the MacPDF sources.

Reply:
I'll merge this and let someone with more domain knowledge tidy it up :P